contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_valid_iata_code.py

```diff
@@ -0,0 +1,138 @@
+from typing import Optional
+from urllib.error import HTTPError, URLError
+from urllib.request import Request, urlopen
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import PandasExecutionEngine
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+BASE_API_URL = "https://content.airhex.com/api/v3.7.4/airports/?iata="
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesToBeValidIataCode(ColumnMapMetricProvider):
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.valid_iata_code"
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, **kwargs):
+        def valid_iata_code(x):
+            if not isinstance(x, str):
+                return False
+            if len(x) != 3:
+                return False
+            try:
+                urlopen(
+                    Request(
+                        BASE_API_URL + x, headers={"User-Agent": "great_expectations"}
+                    )
+                )
+                return True
+            except (URLError, HTTPError):
+                return False
+
+        return column.apply(lambda x: valid_iata_code(x) if x else False)
+
+    # This method defines the business logic for evaluating your metric when using a SqlAlchemyExecutionEngine
+    # @column_condition_partial(engine=SqlAlchemyExecutionEngine)
+    # def _sqlalchemy(cls, column, _dialect, **kwargs):
+    #     raise NotImplementedError
+
+    # This method defines the business logic for evaluating your metric when using a SparkDFExecutionEngine
+    # @column_condition_partial(engine=SparkDFExecutionEngine)
+    # def _spark(cls, column, **kwargs):
+    #     raise NotImplementedError
+
+
+# This class defines the Expectation itself
+class ExpectColumnValuesToBeValidIataCode(ColumnMapExpectation):
+    """This Expectation validates data as conforming to the valid IATA airport codes."""
+
+    # These examples will be shown in the public gallery.
+    # They will also be executed as unit tests for your Expectation.
+    examples = [
+        {
+            "data": {
+                "valid_users": ["MCO", "SFO", "LAX", "JFK", "LGA", "EWR"],

```

Comment:
Maybe rename these to `valid_codes` and `invalid_codes`

Reply:
Good catch. Updated and committed.